compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -139,15 +139,15 @@ fn parse_args<'a>(ecx: &mut ExtCtxt<'a>, sp: Span, tts: TokenStream) -> PResult<
             _ => {
                 let expr = p.parse_expr()?;
                 if !args.named_args().is_empty() {
-                    ecx.dcx().emit_err(errors::PositionalAfterNamed {
```

Comment:
This seems unconnected with the second half of this commit. Is it necessary?

Also, if necessary, it seems incomplete, because there are two other emit calls in this function. Seems like they should be converted to return created errors as well?

Reply:
The others return. this one doesn't

Reply:
I see one `err.emit()` and two `emit_err()` calls in `parse_args`. Prior to this change, none of them returned. After this change, one of them returns an `Err` without emitting, and the other two are unchanged.

Reply:
Oh hmm. Need to rethink this one

Reply:
The reason I did this is so that expand_format_args_impl (which is the only `parse_args` caller), doesn't cause follow-up warnings like

Reply:
I'm now not clear if you're intending to do more work here ("need to rethink this one") or if you are waiting for a response from me. Can you clarify?

Reply:
There are no changes needed. this early return ensures that the macro expansion is poisoned instead of expanding to nonsense that may cause follow up nonsensical diagnostics.

Reply:
Basically I think we should merge it as is, but I can also remove the entire commit and file it as a separate PR if you prefer

Reply:
Ok, I'll accept that explanation, thanks.